[BUG] glib 0.14.3 fails to typecheck / compile on non-x86 architectures

Version 0.14.3 of the `glib` crate does not compile on any non-x86 architecture (aarch64, armv7hfp, powerpc64le, s390x).

This commit seems to have introduced code that does not typecheck correctly on non-x86 architectures:
https://github.com/gtk-rs/gtk-rs-core/commit/9b9d47876a5898f5d9c756f8ea0f09bf5c54a204

This is the compiler error message:

```
error[E0308]: mismatched types
  --> src/variant_iter.rs:88:17
   |
88 |                 s as *const u8 as *const i8,
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`
   |
   = note: expected raw pointer `*const u8`
              found raw pointer `*const i8`
error: aborting due to previous error
For more information about this error, try `rustc --explain E0308`.
error: could not compile `glib`
```

--- a/glib/src/variant_iter.rs
+++ b/glib/src/variant_iter.rs
@@ -85,7 +85,7 @@ impl<'a> VariantStrIter<'a> {
             ffi::g_variant_get_child(
                 self.variant.to_glib_none().0,
                 i,
-                s as *const u8 as *const i8,
+                s as *const u8 as *const _,
                 &p,
                 std::ptr::null::<i8>(),
             );

